### DISCUSSION TITLE
Packages as layers in another language

### DISCUSSION THREAD

**Discussion by wale-paystack** [benbjohnson/wtf#45] [Q&A] [answered]
hey @benbjohnson - thank you for creating this sample repo to help illustrate the ideas presented in your standard package layout article - https://www.gobeyond.dev/standard-package-layout/. I stumbled upon it when I started learning Go, and I've been ever thankful. it has helped me internalize the mechanism fully. 

I'm just curious to know if you apply this same approach when building in another language like say TypeScript/js with Nodejs. I know how you group your code in Nodejs and other programming languages mostly comes down to personal preference, and a standard doesn't really matter just like you said in a recent article but was just curious.

**Comment by benbjohnson** [comment, marked as answer]
Hi @wale-paystack. I'm glad to hear it's been helpful. Thanks for reaching out.

Honestly, I haven't done many medium or large-sized projects in other languages in a while. I find many other languages like JavaScript or Ruby tend to have popular frameworks that dictate the directory layout so I tend to stick to what those frameworks recommend. However, I find that I still try to bring [the layering mindset](https://www.gobeyond.dev/packages-as-layers/) to any code I write.

**Reply by wale-paystack** [reply]
Thank you.
I've carried this thinking with me to new (Go) projects since I learned about it. I recently started a new Node.js (TypeScript) project with the same mindset, hence my curiosity to know if it is something you've done.
